### PR TITLE
Fixed #632: infinite scrolling problem on firefox

### LINKED ISSUE #632: infinite scrolling doesn't work in Firefox
Infinite scrolling of messages in bout doesn't work in Firefox. I'm not sure about bouts in inbox, but I suspect that it also doesn't work.

### PR DESCRIPTION
**Pull request by erimerturk**
Fixed infinite scrolling problem on inbox and bout page for firefox #632

### CODE REVIEW

**Comment by karato** [comment]
@erimerturk Thanks, I will find someone to review this PR soon

**Comment by karato** [comment]
@pinaf review it

File: netbout-web/src/main/js/bout.js [hunk 1]
```diff
@@ -69,31 +69,51 @@ $(document).ready(
               dataType: 'xml',
               method: 'GET',
               success: function (data) {
-                var appendix = '';
+                var appendix = '', $data = $(data), messagesAsXML = $data.find('message'),
```
Comment (pinaf) [inline netbout-web/src/main/js/bout.js]: @erimerturk can we rename `messagesAsXML` to `msgs` instead?

File: netbout-web/src/main/js/bout.js [hunk 2]
```diff
@@ -69,31 +69,51 @@ $(document).ready(
...
+                    }
+                  );
+                  $tail.removeAttr('id');
+                  $tail.html(appendix + '<div id="tail"/>');
+                  $box.attr('data-more', more);
+                }
+                if (messages) {
+                  messages.find('.message').each(
+                    function (idx, line) {
+                      var temp = $('<div></div>'), $msg = $(line),
```
Comment (pinaf) [inline netbout-web/src/main/js/bout.js]: @erimerturk Let's avoid declaring multiple variables on the same line as it makes the code hard to read.

File: netbout-web/src/main/js/bout.js (hunk 1 above)
Comment (pinaf) [inline netbout-web/src/main/js/bout.js]: @erimerturk Let's avoid declaring multiple variables on the same line as it makes the code hard to read.

File: netbout-web/src/main/js/bout.js (hunk 2 above)
Comment (pinaf) [inline netbout-web/src/main/js/bout.js]: @erimerturk can't we inline `temp`?

File: netbout-web/src/main/js/inbox.js [hunk 3]
```diff
@@ -51,48 +51,61 @@ $(document).ready(
               dataType: 'xml',
               method: 'GET',
               success: function (data) {
-                var appendix = '<ul class="bouts">';
+                var appendix = '<ul class="bouts">', $data = $(data),
```
Comment (pinaf) [inline netbout-web/src/main/js/inbox.js]: @erimerturk Let's avoid declaring multiple variables on the same line as it makes the code hard to read.

File: netbout-web/src/main/js/inbox.js [hunk 4]
```diff
@@ -51,48 +51,61 @@ $(document).ready(
...
-                    );
-                    appendix += '</div></li>';
-                    more = $bout.find('link[rel="more"]').attr('href');
+                if (boutsAsXml) {
+                  boutsAsXml.find('bout').each(
+                    function (idx, bout) {
+                      var $bout = $(bout),
+                          unread = parseInt($bout.find('unread').text(), 10),
+                          unseen = parseInt($bout.find('unseen').text(), 10);
+                      appendix += [
```
Comment (pinaf) [inline netbout-web/src/main/js/inbox.js]: @erimerturk let's extract this code (up tp line 78) into a separate function with a good name

File: netbout-web/src/main/js/inbox.js [hunk 5]
```diff
@@ -51,48 +51,61 @@ $(document).ready(
...
+                        '">',
+                        escapeHTML($bout.find('title').text()),
+                        '</a>',
+                        (unread === 0 ? '' : '<span class="unread">' + unread + '</span>'),
+                        '</h1><div class="friends">'
+                      ].join('');
+                      $bout.find('friend').each(
+                        function (idx, friend) {
+                          var $friend = $(friend), shift = 57 * idx;
+                          appendix += [
```
Comment (pinaf) [inline netbout-web/src/main/js/inbox.js]: @erimerturk let's extract this code (up tp line 89) into a separate function with a good name

File: netbout-web/src/main/js/bout.js [hunk 6]
```diff
@@ -69,31 +69,51 @@ $(document).ready(
...
-                  }
-                );
-                $tail.removeAttr('id');
-                $tail.html(appendix + '<div id="tail"/>');
-                $box.attr('data-more', more);
+                if (messagesAsXML) {
+                  messagesAsXML.each(
+                    function (idx, msg) {
+                      var $msg = $(msg);
+                      appendix += [
```
Comment (pinaf) [inline netbout-web/src/main/js/bout.js]: @erimerturk let's extract this code (up to line 92) into a separate function with a good name

**Comment by pinaf** [comment]
@erimerturk 8 comments above

**Comment by pinaf** [comment]
@erimerturk let's avoid declaring multiple variables on the same line

**Comment by pinaf** [comment]
@erimerturk also, some of the code that appends a string to a variable called appendix is very hard to read. let's move the string to be appended into a separate function with a decent name in order to make the code more readable.

**Comment by pinaf** [comment]
@erimerturk also, let's avoid composite variable names.

**Comment by erimerturk** [comment]
@pinaf please review it, thank you

**Comment by pinaf** [comment]
@erimerturk much better. can we rename `asXml` to `xml` and `asHtml` to `html`?

**Comment by erimerturk** [comment]
@pinaf could you please review it, thank you

**Comment by pinaf** [comment]
@erimerturk thank you

**Comment by pinaf** [comment]
@rultor merge

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by dmzaytsev** [comment]
@rultor deploy

**Comment by karato** [comment]
@elenavolokhova please, review this ticket for compliance with our [QA rules](http://www.teamed.io/qa.html)

**Comment by karato** [comment]
@rultor deploy pls

**Comment by elenavolokhova** [comment]
@karato Quality is good here.

**Comment by karato** [comment]
> @karato Quality is good here.

@elenavolokhova thank you, good quality is always good ;)

**Comment by karato** [comment]
@pinaf I added 10 mins to @elenavolokhova (for QA review) in transaction 59118181

since quality is good, 10 mins paid to @dmzaytsev (our architect), transaction `59118184`

Thanks for your contribution, **22 mins** was added to your account, payment ID is `AP-7FR26717KC870023N`, 104 hours and 54 mins spent total

you're getting extra minutes here (c=7)

added +22 to your rating, now it is equal to [+6896](http://www.netbout.com/b/35128?open=rating)

**Pull request merged**